packages/react-button/src/components/Button/useButtonStyles.ts

```diff
@@ -261,29 +278,34 @@ export const useButtonStyles = (state: ButtonState) => {
   const styles = useStyles();
   state.className = mergeClasses(
     styles.root,
-    state.iconOnly && styles.rootIconOnly,
     state.primary && styles.rootPrimary,
     state.subtle && styles.rootSubtle,
     state.transparent && styles.rootTransparent,
     state.disabled && styles.rootDisabled,
     state.disabled && state.primary && styles.rootDisabledPrimary,
     state.disabled && state.subtle && styles.rootDisabledSubtle,
     state.disabled && state.transparent && styles.rootDisabledTransparent,
-    state.size === 'small' && styles.rootSmall,
-    state.size === 'large' && styles.rootLarge,
-    state.iconOnly && state.size === 'small' && styles.rootIconOnlySmall,
-    state.iconOnly && state.size === 'large' && styles.rootIconOnlyLarge,
+    styles[`root${capitalizeString(state.size)}` as `root${Capitalize<ButtonState['size']>}`],
+    state.iconOnly &&
+      styles[`rootIconOnly${capitalizeString(state.size)}` as `rootIconOnly${Capitalize<ButtonState['size']>}`],
```

Comment:
You should rename the styles to match the enum values (`rootLarge` => `large`), and not concatenate/capitalize the enum value. Then you can simply write `styles[state.size]`.

And for styles that rely on more than one prop (like iconOnly + size), I'd leave it as it was, rather than doing concatenation and capitalization. Or alternatively, you could make a separate style object (separate call to makeStyles) called e.g. useIconOnlyStyles, with styles named `small`, `medium`, and `large`. Then here you could do `iconOnlyStyles[state.size]`. However, I'm not sure if there are perf concerns with adding extra calls to makeStyles/useStyles.

The main benefit of using the prop value directly to pick the style is simplicity and type checking, but this kind of defeats that simplicity.

Reply:
Ben and I had an offline conversation about this and I have made some changes to better adhere to what we said there.